src/Settings/index.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { FunctionComponent } from "react"
+import { ScrollView, View, StyleSheet, TouchableOpacity } from "react-native"
+import { useTranslation } from "react-i18next"
+import { SvgXml } from "react-native-svg"
+import { useNavigation } from "@react-navigation/native"
+import env from "react-native-config"
+
+import { getLocalNames } from "../locales/languages"
+import { GlobalText } from "../components"
+import {
+  useStatusBarEffect,
+  Stacks,
+  ModalScreens,
+  SettingsScreens,
+} from "../navigation"
+import { useConfigurationContext } from "../ConfigurationContext"
+
+import { Icons } from "../assets"
+import { Iconography, Colors, Spacing, Typography, Outlines } from "../styles"
+
+const Settings: FunctionComponent = () => {
+  useStatusBarEffect("light-content", Colors.headerBackground)
+  const navigation = useNavigation()
+  const {
+    t,
+    i18n: { language: localeCode },
+  } = useTranslation()
+  const languageName = getLocalNames()[localeCode]
+  const {
+    displayCallbackForm,
+    displayReportAnIssue,
+  } = useConfigurationContext()
+  const showDebugMenu = env.STAGING === "true" || __DEV__
+
+  const handleOnPressSelectLanguage = () => {
+    navigation.navigate(Stacks.Modal, {
+      screen: ModalScreens.LanguageSelection,
+    })
+  }
+
+  type SettingsListItem = {
+    label: string
+    onPress: () => void
+  }
+
+  const About: SettingsListItem = {
+    label: t("screen_titles.about"),
+    onPress: () => navigation.navigate(SettingsScreens.About),
+  }
+  const Legal: SettingsListItem = {
+    label: t("screen_titles.legal"),
+    onPress: () => navigation.navigate(SettingsScreens.Legal),
+  }
+  const CallbackForm: SettingsListItem = {
+    label: t("screen_titles.callback_form"),
+    onPress: () => navigation.navigate(SettingsScreens.CallbackForm),
+  }
+  const ReportAnIssue: SettingsListItem = {
+    label: t("screen_titles.report_issue"),
+    onPress: () => navigation.navigate(SettingsScreens.ReportIssue),
+  }
+
+  const baseListItems: SettingsListItem[] = [About, Legal]
+
+  const settingsListItems: (SettingsListItem | undefined)[] = [
+    ...baseListItems,
+    displayCallbackForm ? CallbackForm : undefined,
+    displayReportAnIssue ? ReportAnIssue : undefined,
+  ]
+
+  const settingsListItemsToDisplay: SettingsListItem[] = settingsListItems.filter(
```

Comment:
I dig the functional style! though i think you could get away without the filter if you hoist the conditionals up and allow for some mutation. e.g.
```typescript
const listItems = [About, Legal]
if (displayCallbackForm) {
  listItems.push(CallbackForm)
}
if (displayReportAnIssue) {
  listItems.push(ReportAnIssue)
}
```
which might be a bit easier to follow.

Reply:
Ahhh this is much easier to follow, thanks!

Reply:
Love the nerdy programming thingz!